src/coreclr/vm/interpexec.cpp

```diff
@@ -1982,27 +1992,42 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                         if (targetIp == NULL)
                         {
                             // If we didn't get the interpreter code pointer setup, then this is a method we need to invoke as a compiled method.
+                            // Interpreter-FIXME: Implement tailcall via helpers, see https://github.com/dotnet/runtime/blob/main/docs/design/features/tailcalls-with-helpers.md
```

Comment:
I don't really understand this comment. If we have a tailcall that needs to transition from interpreter to compile code, it doesn't seem like it makes sense to do a tail call with assembly magic. I see at most we could subtract the interpreter stack pointer before doing the call.

Reply:
Tailcalls via helpers scheme is used by the JIT to implement tailcalls between methods where fast tail calls are not possible. For example, `void M1(a few arguments) { tail call M2(...); } void M2(a lot of arguments) { tail call M1(...); }`. It is not possible to tailcall M2 from M1 since there is not enough stack space to pass the arguments. The scheme works by injecting an extra frame where we run the tailcalls in a loop without consuming stack space. The design doc goes into details that I won't repeat here.

I believe that the same scheme should be able to handle the situation like `void M1_JITed() { tail call M2_Interpreted(...); } void M2_Interpreted() { tail call M1_JITed(...); }` without causing stackoverflow.

Reply:
Re: Subtracting the stack pointer, I believe that shouldn't be necessary since the stack is supposed to be empty (other than arguments) at the time of the tail call, and the tail call takes ownership of the arguments. The insn after the tail call is required to be a ret. Or am I misunderstanding and I need to do something extra?